File: public/examples/integrations/mcp-servers/trello_api/get_trello_action_example_call_tool.py
import json
from arcadepy import Arcade

client = Arcade()  # Automatically finds the `ARCADE_API_KEY` env variable

USER_ID = "{arcade_user_id}"
TOOL_NAME = "TrelloApi.GetTrelloAction"

auth_response = client.tools.authorize(
    tool_name=TOOL_NAME,
    user_id=USER_ID,
)

if auth_response.status != "completed":
    print(f"Click this link to authorize: {auth_response.url}")

# Wait for the authorization to complete
client.auth.wait_for_completion(auth_response)

tool_input = {
    'action_id': '64b8f2a3e1c4b12d9f0a7b21',
    'action_fields': 'type,date,data,memberCreator',
    'member_fields_list': 'id,username,fullName',
    'member_creator_fields': 'all',
    'include_display': True,
    'include_entities': True,
    'include_member': False,
    'include_action_creator': True
}

response = client.tools.execute(
    tool_name=TOOL_NAME,
    input=tool_input,
    user_id=USER_ID,
)
print(json.dumps(response.output.value, indent=2))

File: public/examples/integrations/mcp-servers/trello_api/get_trello_action_example_call_tool.js
import { Arcade } from "@arcadeai/arcadejs";

const client = new Arcade(); // Automatically finds the `ARCADE_API_KEY` env variable

const USER_ID = "{arcade_user_id}";
const TOOL_NAME = "TrelloApi.GetTrelloAction";

// Start the authorization process
const authResponse = await client.tools.authorize({tool_name: TOOL_NAME});

if (authResponse.status !== "completed") {
  console.log(`Click this link to authorize: ${authResponse.url}`);
}

// Wait for the authorization to complete
await client.auth.waitForCompletion(authResponse);

const toolInput = {
  "action_id": "64b8f2a3e1c4b12d9f0a7b21",
  "action_fields": "type,date,data,memberCreator",
  "member_fields_list": "id,username,fullName",
  "member_creator_fields": "all",
  "include_display": true,
  "include_entities": true,
  "include_member": false,
  "include_action_creator": true
};

const response = await client.tools.execute({
  tool_name: TOOL_NAME,
  input: toolInput,
  user_id: USER_ID,
});

console.log(JSON.stringify(response.output.value, null, 2));
